### ISSUE TITLE
Full import/export of all data

### ISSUE THREAD

**Issue by ostcar** [OpenSlides/OpenSlides#434]
We need a funktion to export and import all data, incl. permissions and (hash) passwords.

**Comment by emanuelschuetze** [comment]
Discussed with @FinnStutzenstein and @boehlke :
Try to use [dumpdata](https://docs.djangoproject.com/en/dev/ref/django-admin/#dumpdata) and [loaddata](https://docs.djangoproject.com/en/dev/ref/django-admin/#django-admin-loaddata) by Django for full export and import.

**Comment by ostcar** [comment]
@emanuelschuetze could you tell me once more what do you want? We have the function `element_cache.get_all_full_data()` which returns a dict of all data in json.

You can access (nearly) this data, if you connect to openslides via an admin user on websocket.

So if you just want a json file of all data, we already have all necessary functions.

**Comment by jsaalfeld** [comment]
Currently we just dump all the sql data and the static files. With the help of that we can recreate instances immediately. Is there a need of other functions?

**Comment by ostcar** [comment]
no. It is not possible to import get_all_full_data. Maybe it will be, when we switch to redis or another nosql backend.

I don't understand what the goal of this issue is. If you want to transfer the date from one instance to another I would suggest a sqldump.

I thought emanuel wanted a way to export the data so it can be imported into other tools like exel.

**Comment by emanuelschuetze** [comment]
The goal of this issue: A user with admin permissions in OpenSlides (not a database admin!) should get the possibility to export all data for backup via web interface - and import all data in a new instance.
Currently the user can only export some modules via csv (but csv is a bad format to make references, e.g. submitters in motions or save all change recommendations, personal notes etc.). Currently you cannot _restore_ with some csv export a full instance. You need a sqldump.

**Comment by emanuelschuetze** [comment]
Maybe with history mode we can solve some of the goals (backup, partly and manually restore).

Manually export a motion (with all metadata etc.) as json for full import could be useful.

**Comment by Hackwar** [comment]
I want to second the idea of a complete im- and export. My specific situation today was, that I prepared a raspi with OpenSlides and added all the motions, users, etc. to it and the idea was, that the raspi was connected to the projector and provided its own wifi network. So in theory I only had to plug in the power to the raspi, the HDMI to the projector and optionally the ethernet and I could connect with practically everything to the wifi and start the conference. 

Unfortunately the projector only had VGA input :see_no_evil: :speak_no_evil: I had asked to make sure that the projector had HDMI... So now I was sitting there and couldn't really use the raspi and thus had to start from scratch half an hour before the event started. If I had an im/export, I could have exported from the raspi and then imported into my laptops installation. I know that I could have used the raspi as server and my laptop as projector and for management, but with all the issues that I had prior to that situation, I lost trust in the solution and rather wanted to start from scratch...

**Comment by ostcar** [comment]
This issue is about exporting the data out of OpenSlides, to use it from another application.

@Hackwar, you have a different issue. You want to export all data from one OpenSlides instance to start another OpenSlides instance. This is already possible and I am sorry, that you had to re-add all the data manually.

OpenSlides uses per default an sqlite database, therefore one file. In the settings is the path of this database-file. You can copy this file from one machine (raspi) to another (laptop) and start just start OpenSlides. If you use postgres or mysql, you can get the same result with an sqldump.

The only problem could be media-files. But if media-files are really a problem, then please open another issue with an explanation what happens with them, if you move them from one instance to another.

**Comment by Hackwar** [comment]
Sorry for disagreeing here, but the issue is slightly different. I used the ansible script to prepare the raspi and that installs a postgres instance. However when everything failed, I went back to the portable version on my windows machine, so now I would have had to export from postgres into sqlite and all that under a bit of pressure. I know my computers a fair bit, but even I wouldn't know how to do that, especially not in a situation where it has to happen fast. Considering that I expect tech unsavvy people to handle this in the future, I would rather have a (custom) export format/feature where I can click on export to get all data and then on import in another instance.
